### PR TITLE
Retrying to connect HTTP/S servers

### PR DESCRIPTION
**Pull request by pinebit**

### CODE REVIEW

File: core/cmd/client.go [hunk 1]
```diff
@@ -270,6 +270,23 @@ func (n ChainlinkRunner) Run(ctx context.Context, app chainlink.Application) err
 	return errors.WithStack(g.Wait())
 }
 
+func logErrorAndRepeat(ctx context.Context, lggr logger.Logger, cfg config.GeneralConfig, fn func() error) {
+	for {
+		// try calling fn() and log error if any
+		if err := fn(); err != nil {
+			lggr.Criticalf("%v", err)
```
Comment (samsondav) [inline core/cmd/client.go:277]: What about the expected ErrServerClosed on a clean shutdown?

Comment (pinebit) [reply core/cmd/client.go]: Fixed

File: core/cmd/client.go [hunk 2]
```diff
@@ -270,6 +270,25 @@ func (n ChainlinkRunner) Run(ctx context.Context, app chainlink.Application) err
 	return errors.WithStack(g.Wait())
 }
 
+func tryRunServerUntilCancelled(ctx context.Context, lggr logger.Logger, cfg config.GeneralConfig, runServer func() error) {
+	for {
+		// try calling runServer() and log error if any
+		if err := runServer(); err != nil {
+			if err != http.ErrServerClosed {
+				lggr.Criticalf("%v", err)
+			}
```
Comment (samsondav) [inline core/cmd/client.go:279]: Can ErrServerClosed ever happen unexpectedly and signify any breakage? Or is it only on clean shutdown?

Comment (pinebit) [reply core/cmd/client.go:279]: in my experience this error is returned from both ListenAndServe() and Shutdown()

Comment (pinebit) [reply core/cmd/client.go:279]: Also, based on the spec:
```
// ErrServerClosed is returned by the Server's Serve, ServeTLS, ListenAndServe,
// and ListenAndServeTLS methods after a call to Shutdown or Close.
```
it should not happen in other cases

**Review by samsondav — APPROVED** [review]

File: core/cmd/client.go (hunk 2 above)
Comment (jmank88) [reply core/cmd/client.go:279]: Are the logs in `run()` and `runTLS()` redundant now that we log here?
Should this log have a message prefix before the error?

Comment (jmank88) [reply core/cmd/client.go:279]: (changing in the open PR #5929 too)

Comment (pinebit) [reply core/cmd/client.go:279]: Fixed

**Review by jmank88 — APPROVED** [review]

**Pull request merged**